### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by Shevtsooov**

### CODE REVIEW

File: src/validateURL.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+/* eslint-disable max-len */
+const supportedCases = ['SNAKE', 'KEBAB', 'CAMEL', 'PASCAL', 'UPPER'];
+
+const errors = {
+  noText:
+    'Text to convert is required. Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".',
+  noToCase:
```
Comment (33Orange) [inline src/validateURL.js:7]:
```suggestion
  missingText:
    'Text to convert is required. Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".',
  missingCase:
```

File: src/validateURL.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+      ...!toCase
+        ? [{ message: errors.noToCase }]
+        : [],
+
+      ...!supportedCases.includes(toCase) && toCase
+        ? [{ message: errors.notSupported }]
+        : [],
+    ],
+  };
```
Comment (33Orange) [inline src/validateURL.js:28]: It's not very readable

You used so many arrays just to write this logic in ternary operators. 
Please rewrite it in default `if` statements.

```suggestion
  const errors = []
  
  if (condition) {
    errors.push(smth)
  }
  if (condition) {
    errors.push(smth)
  }
  
  return errors;
```

File: src/validateURL.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
+/* eslint-disable max-len */
+const supportedCases = ['SNAKE', 'KEBAB', 'CAMEL', 'PASCAL', 'UPPER'];
+
+const errors = {
```
Comment (33Orange) [inline src/validateURL.js:4]: Use more descriptive names.
```suggestion
const errorsMessages = {
```

File: src/createServer.js [hunk 4]
```diff
@@ -1,3 +1,36 @@
...
+  return http.createServer((req, res) => {
+    res.setHeader('Content-Type', 'application/json');
+
+    const normalizedURL = new URL(req.url, `http://${req.headers.host}`);
+    const toCase = normalizedURL.searchParams.get('toCase');
+    const textToConvert = normalizedURL.pathname.slice(1);
+
+    const errorMessage = validateURL(toCase, textToConvert);
+
+    if (errorMessage.errors.length) {
```
Comment (33Orange) [inline src/createServer.js:15]:
```suggestion
    if (errors.length) {
```

File: src/createServer.js [hunk 5]
```diff
@@ -1,3 +1,36 @@
...
+
+      const transformedText = convertToCase(textToConvert, toCase);
+      const data = {
+        ...transformedText,
+        targetCase: toCase,
+        originalText: textToConvert,
+      };
+
+      res.end(JSON.stringify(data));
+    }
```
Comment (33Orange) [inline src/createServer.js:30]: Is it necessary to wrap the default state in `else` construction?

```suggestion
    } 
    res.statusCode = 200;

    const transformedText = convertToCase(textToConvert, toCase);
    const data = {
      ...transformedText,
      targetCase: toCase,
      originalText: textToConvert,
    };

    res.end(JSON.stringify(data));
```

File: src/createServer.js [hunk 6]
```diff
@@ -1,3 +1,36 @@
...
+      res.end(JSON.stringify(errorMessage));
+    } else {
+      res.statusCode = 200;
+
+      const transformedText = convertToCase(textToConvert, toCase);
+      const data = {
+        ...transformedText,
+        targetCase: toCase,
+        originalText: textToConvert,
+      };
```
Comment (33Orange) [inline src/createServer.js:27]: image


```suggestion
      const { originalCase, convertedText } = convertToCase(textToConvert, toCase);
      
      const data = {
        originalCase,
        targetCase: toCase,
        originalText: textToConvert,
        convertedText,
      };
```

File: src/createServer.js [hunk 7]
```diff
@@ -1,3 +1,36 @@
...
+    const textToConvert = normalizedURL.pathname.slice(1);
+
+    const errorMessage = validateURL(toCase, textToConvert);
+
+    if (errorMessage.errors.length) {
+      res.statusCode = 400;
+      res.statusMessage = 'Bad request';
+      res.end(JSON.stringify(errorMessage));
+    } else {
+      res.statusCode = 200;
```
Comment (33Orange) [inline src/createServer.js:20]: `You should respond with status 200 and OK status text.`

Check the task requirements, please

**Comment by Shevtsooov** [comment]
@33Orange Hi, thanks!
Fixed the mentioned issues.

File: src/validateURL.js [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+const supportedCases = ['SNAKE', 'KEBAB', 'CAMEL', 'PASCAL', 'UPPER'];
+
+const errorMessages = {
+  missingText:
+    'Text to convert is required. Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".',
+  missingCase:
+    '"toCase" query param is required. Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".',
+  notSupported:
+    `This case is not supported. Available cases: ${supportedCases.join(', ')}.`,
+};
```
Comment (silchyn) [inline src/validateURL.js:11]: Could be in a separate file like `constants.js`

File: src/createServer.js [hunk 9]
```diff
@@ -1,3 +1,44 @@
...
+
+const createServer = () => {
+  return http.createServer((req, res) => {
+    res.setHeader('Content-Type', 'application/json');
+
+    const normalizedURL = new URL(req.url, `http://${req.headers.host}`);
+    const toCase = normalizedURL.searchParams.get('toCase');
+    const textToConvert = normalizedURL.pathname.slice(1);
+
+    const error = validateURL(toCase, textToConvert);
```
Comment (silchyn) [inline src/createServer.js:13]:
```suggestion
    const errors = validateURL(toCase, textToConvert);
```

File: src/createServer.js [hunk 10]
```diff
@@ -1,3 +1,44 @@
...
+
+    res.statusCode = 200;
+    res.statusMessage = 'OK';
+
+    const {
+      originalCase,
+      convertedText,
+    } = convertToCase(textToConvert, toCase);
+
+    const data = {
```
Comment (silchyn) [inline src/createServer.js:31]: That's redundant

**Review by silchyn — APPROVED** [review]

**Pull request closed without merging**